src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -313,6 +335,25 @@ export class KernelDebugAdapter implements DebugAdapter, IKernelDebugAdapter {
             }
         });
 
+        if ((message as DebugProtocol.StackTraceResponse).command === 'stackTrace') {
+            (message as DebugProtocol.StackTraceResponse).body.stackFrames.forEach((sf) => {
+                // Check if we're stopped at the last line
+                let currentCell: NotebookCell | undefined;
+                this.notebookDocument.getCells().forEach((cell) => {
+                    const index = sf.source?.path!.indexOf('#ch');
```

Comment:
This code can go outside the forEach loop, this doesn't change, hence no need to get index & fragment everytime for each cell.